deploy/helm/elastic-agent/templates/integrations/_kubernetes/_kubernetes_logs_containers.tpl

```diff
@@ -16,12 +16,12 @@ Config input for container logs
     namespace: {{ .Values.kubernetes.namespace }}
   use_output: {{ .Values.kubernetes.output }}
   streams:
-  - id: kubernetes-container-logs-${kubernetes.pod.name}-${kubernetes.container.id}
+  - id: kubernetes-container-logs-${kubernetes.namespace}-${kubernetes.pod.name}-${kubernetes.container.name}
```

Comment:
The new ID does not seem to be unique, this will cause problems and lead to data duplication.
Are you sure that 
`${kubernetes.namespace}-${kubernetes.pod.name}-${kubernetes.container.name}` will always be unique and will make this block be rendered only once?

The previous implementation used the `${kubernetes.container.id}` because that's guaranteed to always be unique.
```suggestion
  - id: kubernetes-container-logs-${kubernetes.pod.name}-${kubernetes.container.id}
```

The key thing we have to ensure here is that there will never be two Filestream inputs started with the same ID.

Reply:
I see this file is in the helm chart folder, how does upgrade here work? Is there any change an user will upgrade and have the input ID changing? If so, that will lead to data duplication because the new ID will make all files be considered new, thus fully re-ingested.

Reply:
@belimawr some things I have in mind, the combination of `${kubernetes.namespace}-${kubernetes.pod.name}-${kubernetes.container.name}` should always be unique in a k8s cluster, right?! 

also about the 
> how does upgrade here work

indeed when we restart we might re-ingest the logs but if we have to do it, we should do it now that helm chart is not GA yet. 🙂

Reply:
> the combination of ${kubernetes.namespace}-${kubernetes.pod.name}-${kubernetes.container.name} should always be unique in a k8s cluster, right?!

I hope so, however I'm not sure. It would be nice to back it with some K8s documentation. I know for sure `${kubernetes.container.id}` is always unique, hence it's the one I've always used/recommended.

> indeed when we restart shouldn't fingerprint be the same no matter the ID?!

Reply:
Quoting from k8s docs https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#when-to-use-multiple-namespaces

> Namespaces provide a scope for names. Names of resources need to be unique within a namespace, but not across namespaces.
hence, given a namespace + podname pair it's guaranteed to be unique in the cluster (as opposed to the previous id that was NOT namespaced)

Within a pod spec container names must be unique so once the pod is created the tuple (namespace, pod name, container name) is unique and the input definition does not change with container restarts.

Reply:
Thanks for checking the docs and clarifying @pchila !!

Then ignore the change I suggested.

The only issue that persists is the ID changing when the helm chart is upgraded... That will cause re-ingestion of files, even with fingerprint.